src/curl_async.c

```diff
@@ -157,6 +158,11 @@ static void free_curl_file(struct swupd_curl_parallel_handle *h, struct multi_cu
 		curl_easy_cleanup(curl);
 		file->curl = NULL;
 	}
+
+	if (file->progress != NULL) {
```

Comment:
you don't need to check for null before free. if pointer is null free() doesn't fail.

Reply:
nice! thanks for the tip, I'll remove the if